File: 07_Trees/08_Binary_Tree_Level_Order_Traversal.py
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        q = []
        ans = []

        if root:
            q.append(root)
        
        while len(q) > 0:
            sz = len(q)
            temp = []
            while sz > 0:
                node = q[0]
                temp.append(q[0].val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
                q.pop(0)
                sz -= 1
            ans.append(temp)

        return ans

File: 07_Trees/08_Binary_Tree_Level_Order_Traversal.cpp
class Solution
{
public:
    vector<vector<int>> levelOrder(TreeNode *root)
    {
        vector<vector<int>> ans;
        queue<TreeNode *> q;

        if (root)
            q.push(root);
        while (!q.empty())
        {
            int sz = q.size();
            vector<int> temp;
            while (sz--)
            {
                TreeNode *node = q.front();
                q.pop();
                temp.push_back(node->val);
                if (node->left)
                    q.push(node->left);
                if (node->right)
                    q.push(node->right);
            }
            ans.push_back(temp);
        }
        return ans;
    }
};